scripts/cat/cats.py

```diff
@@ -0,0 +1,938 @@
+                             self.eye_colour, self.pelt.name)
+        else:
+            self.name = Name(status, prefix, suffix, eyes=self.eye_colour)
+
+        # APPEARANCE
+        init_eyes(self)
+        init_pelt(self)
+        init_sprite(self)
+        init_scars(self)
+        init_accessories(self)
+        init_white_patches(self)
+        init_pattern(self)
+
+        self.paralyzed = False
+        self.no_kits = False
+        self.exiled = False
+        if self.genderalign == None:
+            self.genderalign = self.gender
+
+        # Sprite sizes
+        self.sprite = None
+        self.big_sprite = None
+        self.large_sprite = None
+
+        # experience and current patrol status
+        self.in_camp = 1
+        if self.age in ['kitten']:
+            self.experience = 0
+        elif self.age in ['adolescent']:
+            self.experience = randint(0, 19)
+        elif self.age in ['young adult']:
+            self.experience = randint(10, 40)
+        elif self.age in ['adult']:
+            self.experience = randint(20, 50)
+        elif self.age in ['senior adult']:
+            self.experience = randint(30, 60)
+        elif self.age in ['elder']:
+            self.experience = randint(40, 70)
+        else:
+            self.experience = 0
+
+        self.paralyzed = False
+        self.no_kits = False
+        self.exiled = False
+        if self.genderalign == None:
+            self.genderalign = self.gender
+
+        # SAVE CAT INTO ALL_CATS DICTIONARY IN CATS-CLASS
+        self.all_cats[self.ID] = self
+
+    def __repr__(self):
+        return self.ID
+
+    def is_alive(self):
+        return not self.dead
+
+    def die(self):
+        if self.status == 'leader' and game.clan.leader_lives > 0:
+            return
+        elif self.status == 'leader' and game.clan.leader_lives <= 0:
+            self.dead = True
+            game.clan.leader_lives = 0
+        else:
+            self.dead = True
+
+        if self.mate != None:
+            self.mate = None
+            if type(self.mate) == str:
+                mate = Cat.all_cats.get(self.mate)
+                mate.mate = None
+            elif type(self.mate) == Cat:
+                self.mate.mate = None
+
+        for app in self.apprentice.copy():
+            app.update_mentor()
+        self.update_mentor()
+        game.clan.add_to_starclan(self)
+
+    def status_change(self, new_status):
+        # revealing of traits and skills
+        self.update_traits(new_status)
+        # updates mentors
+        if new_status == 'apprentice':
+            self.update_mentor()
+        elif new_status == 'medicine cat apprentice':
+            self.update_med_mentor()
+        # updates skill
+        if self.skill == '???':
+            if new_status == 'warrior' or self.status == 'warrior' and new_status != 'medicine cat':
+                self.skill = choice(self.skills)
+                self.update_mentor()
```

Comment:
Is it gamebreaking? 
Otherwise, I would suggest opening a bug in discords bug_traking channel. Scribble was working on skills and status I think, this _may_ be fixed with the next PR from scribble.

Reply:
Yes, this is the cause of apprentices aging up but still being listed as an apprentice to their mentor

Reply:
fixed and added test